### ISSUE TITLE
Area code not set: Area code must be set before starting a session.

### ISSUE THREAD

**Issue by woei66** [magento/magento2#1941]
environment: centos 6.5 + nginx + php 5.5.29+ msyql 5.6.26+ magento2(git clone)

got a message like below and do action as it suggested:
Run "bin/magento setup:upgrade" from the Magento root directory.
but got below error message
---
<blockquote>
  [Magento\Framework\Exception\SessionException]
  Area code not set: Area code must be set before starting a session.
  [Magento\Framework\Exception\LocalizedException]
  Area code is not set
</blockquote>
---

the reason that I run the "bin/magento setup:upgrade" is that I got a exception message below:
---
</blockquote>
a:4:{i:0;s:1733:"Please upgrade your database: Run "bin/magento setup:upgrade" from the Magento root directory.
The following modules are outdated:
Magento_CatalogSearch data: current version - none, required version - 2.0.0
Magento_SendFriend data: current version - none, required version - 2.0.0
Magento_Ui data: current version - none, required version - 2.0.0
Magento_Sitemap data: current version - none, required version - 2.0.0
Magento_CatalogRuleConfigurable data: current version - none, required version - 2.0.0
Magento_Swagger data: current version - none, required version - 2.0.0
Magento_Swatches data: current version - none, required version - 2.0.0
Magento_Tax data: current version - none, required version - 2.0.0
Magento_TaxImportExport data: current version - none, required version - 2.0.0
Magento_GiftMessage data: current version - none, required version - 2.0.0
Magento_Translation data: current version - none, required version - 2.0.0
Magento_Shipping data: current version - none, required version - 2.0.0
Magento_Ups data: current version - none, required version - 2.0.0
Magento_UrlRewrite data: current version - none, required version - 2.0.0
Magento_EncryptionKey data: current version - none, required version - 2.0.0
Magento_Usps data: current version - none, required version - 2.0.0
Magento_Variable data: current version - none, required version - 2.0.0
Magento_Version data: current version - none, required version - 2.0.0
Magento_Webapi data: current version - none, required version - 2.0.0
Magento_Weee data: current version - none, required version - 2.0.0
Magento_CatalogWidget data: current version - none, required version - 2.0.0
Magento_Wishlist data: current version - none, required version - 2.0.0";i:1;s:965:"#0 [internal function]: Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#1 /usr/share/nginx/magento2/lib/internal/Magento/Framework/Interception/Interceptor.php(141): call_user_func_array(Array, Array)
#2 /usr/share/nginx/magento2/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#3 /usr/share/nginx/magento2/lib/internal/Magento/Framework/App/Http.php(115): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /usr/share/nginx/magento2/lib/internal/Magento/Framework/App/Bootstrap.php(257): Magento\Framework\App\Http->launch()
#5 /usr/share/nginx/magento2/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main}";s:3:"url";s:14:"/admin123/";s:11:"script_name";s:10:"/index.php";}
</blockquote>

**Comment by mazhalai** [comment]
This is a duplicate of 1529
